lib/util.js

```diff
@@ -535,21 +543,45 @@ function findTypedConstructor(value) {
 
 const getBoxedValue = formatPrimitive.bind(null, stylizeNoColor);
 
+// Creates a subclass and name
+// the constructor as `${clazz} : null prototype`
+function clazzWithNullPrototype(clazz, name) {
+
+  class NullPrototype extends clazz {
+    get [Symbol.toStringTag]() {
+      return '';
+    }
+  }
+  Object.defineProperty(NullPrototype.prototype.constructor, 'name',
+                        { value: `[${name}: null prototype]` });
+
+  return NullPrototype;
+}
+
 function noPrototypeIterator(ctx, value, recurseTimes) {
   let newVal;
   // TODO: Create a Subclass in case there's no prototype and show
   // `null-prototype`.
   if (isSet(value)) {
-    const clazz = Object.getPrototypeOf(value) || Set;
+    const clazz = Object.getPrototypeOf(value) ||
+                  clazzWithNullPrototype(Set, 'Set');
     newVal = new clazz(setValues(value));
   } else if (isMap(value)) {
-    const clazz = Object.getPrototypeOf(value) || Map;
+    const clazz = Object.getPrototypeOf(value) ||
+                  clazzWithNullPrototype(Map, 'Map');
     newVal = new clazz(mapEntries(value));
   } else if (Array.isArray(value)) {
-    const clazz = Object.getPrototypeOf(value) || Array;
+    const clazz = Object.getPrototypeOf(value) ||
+                  clazzWithNullPrototype(Array, 'Array');
     newVal = new clazz(value.length || 0);
   } else if (isTypedArray(value)) {
-    const clazz = findTypedConstructor(value) || Uint8Array;
+    let clazz;
+    if (Object.getPrototypeOf(value)) {
```

Comment:
If the value has a prototype, we do not have to search for it anymore. Therefore it would be best to write this as:

```js
const clazz = Object.getPrototypeOf(value) || findTypedConstructor(value) || clazzWithNullPrototype(Uint8Array, 'Uint8Array')
```

Reply:
If I understand this correctly. If we write like the following:

```
const clazz = Object.getPrototypeOf(value) || findTypedConstructor(value) || clazzWithNullPrototype(Uint8Array, 'Uint8Array')
```

when there is no prototype, then `findTypedConstructor` will definitely return the typed array constructor. So we won't reach `clazzWithNullPrototype` (when there is no prototype) -- missing the subclass stuff.

Reply:
You are correct, `findTypedConstructor` should always return the actual type.

Therefore it should likely be:

```js
let clazz = Object.getPrototypeOf(value);
if (!clazz) {
  const constructor = findTypedConstructor(value);
  clazz = clazzWithNullProrotype(constructor, constructor.name);
}
```